docs/tutorials/mobile/superres.md

```diff
@@ -0,0 +1,606 @@
+        }
+    }
+   ```
+
+4. Add the onDestroy method
+
+   ```kotlin
+    override fun onDestroy() {
+        super.onDestroy()
+        ortEnv.close()
+        ortSession.close()
+    }
+
+   ```
+
+5. Add the updateUI method
+
+   ```kotlin
+    private fun updateUI(result: Result) {
+        outputImage?.setImageBitmap(result.outputBitmap)
+    }
+    ```
+
+6. Add the readModel method
+
+   This method reads the ONNX model from the resources folder.
+
+   ```kotlin
+   private fun readModel(): ByteArray {
+       val modelID = R.raw.pt_super_resolution_op16
+       return resources.openRawResource(modelID).readBytes()
+   }   
+   ```
+
+7. Add a method to read the input image
+
+   This method reads a test image from the assets folder. Currently it reads a fixed image built into the application. The sample will soon be extended to read the image directly from the camera or the camera roll.
+
+   ```kotlin
+   private fun readInputImage(): InputStream {
+       return assets.open("test_superresolution.png")
+   }   
+   ```
+
+8. Add the method to perform inference
+
+   This method is calls the heart of the application: `SuperResPerformer.upscale()`, which is the method that runs inference on the model. The code for this is shown in the next section.
+
+   ```kotlin
+    private fun performSuperResolution(ortSession: OrtSession) {
+        var superResPerformer = SuperResPerformer()
+        var result = superResPerformer.upscale(readInputImage(), ortEnv, ortSession)
+        updateUI(result);
+    }   
+   ```
+
+9. Add the TAG object
+
+   ```kotlin
+   companion object {
+       const val TAG = "ORTSuperResolution"
+   }
+   ```
+
+#### Model inference class code
+
+Create a file called `SuperResPerformer.kt` and add the following snippets of code to it.
+
+1. Add imports
+
+   ```kotlin
+   import ai.onnxruntime.OnnxJavaType
+   import ai.onnxruntime.OrtSession
+   import ai.onnxruntime.OnnxTensor
+   import ai.onnxruntime.OrtEnvironment
+   import android.graphics.Bitmap
+   import android.graphics.BitmapFactory
+   import java.io.InputStream
+   import java.nio.ByteBuffer
+   import java.util.*
+   ```
+
+2. Create a result class
+
+   ```kotlin
+   internal data class Result(
+       var outputBitmap: Bitmap? = null
+   ) {}
+   ```
+
+3. Create the super resolution performer class
+
+   This class and its main function `upscale` are where most of the calls to ONNX Runtime live.
+
+   * The `OrtEnvironment` singleton maintains properties of the environment and configured logging levels
+   * `OnnxTensor.createTensor()` is used to create a tensor made up of the input image bytes, suitable as input to the model
+   * `OnnxJavaType.UINT8` is the data type of the input tensor
+   * `OrtSession.run()` run the inference (prediction) on the model to get the output upscaled image
+
+   ```kotlin
+   internal class SuperResPerformer(
+    private val ortSession: OrtSession
```

Comment:
this code has been updated as well.

Ortsession is removed from SuperResPerformer class.

Reply:
FIxed